xbmc/video/windows/GUIWindowVideoBase.cpp

```diff
@@ -897,10 +897,12 @@ void CGUIWindowVideoBase::GetContextButtons(int itemNumber, CContextButtons &but
         buttons.Add(CONTEXT_BUTTON_PLAY_PARTYMODE, 15216); // Play in Partymode
       }
 
-      //if the item isn't a folder or script, is a member of a list rather than a single item
-      //and we're not on the last element of the list,
-      //then add add either 'play from here' or 'play only this' depending on default behaviour
-      if (!(item->m_bIsFolder || item->IsScript()) && m_vecItems->Size() > 1 && itemNumber < m_vecItems->Size()-1)
+      //if the item isn't a folder or script, is not explicitly marked as not playable,
+      //is a member of a list rather than a single item and we're not on the last element of the list,
+      //then add either 'play from here' or 'play only this' depending on default behaviour
+      if (!(item->m_bIsFolder || item->IsScript())
```

Comment:
I'd suggest doing the opposite. Check if the property exists and is explicitly marked as false. You never know which type of items rely on this functionality. So, setting the isPlayable property to false will allow to disable the context item programmatically from addons while retaining the same behaviour as today for all other items.